dataset-processor-python/data_loader.py

```diff
@@ -31,25 +43,186 @@ class DatasetWriter(DataLoader):
 
     def __init__(self, path, dataset_name):
         super().__init__(path, dataset_name)
+        self.cell_feature_analysis_file = []
+        self.feature_defs_file = []
+        self.dataset_file = {}
+        self.feature_def_keys = []
+        self.discrete_features_dict = {}
+        self.features_data_order = []
+
+    def get_user_inputs(self):
+        self.version = input("Enter the version: ").strip()
+        if not self.version or not self.is_valid_version(self.version):
+            print("Please enter a valid version in the format [yyyy.number].")
+            return self.get_user_inputs()
+        self.title = input("Enter the title: ").strip()
+        self.description = input("Enter the description: ").strip()
 
-    def get_inputs(self):
+    def is_valid_version(self, version):
+        pattern = r"^\d{4}\.[0-9]+$"
+        return re.match(pattern, version) is not None
+
+    def process_data(self):
         """
-        get the inputs from the user
+        Process each row of the csv file and compile the json files
         """
-        # version is required, title and description are optional, default values are set as empty string
-        # return version, title, description
-        pass
+        for row in self.data:
+            file_info, features = self.get_row_data(row)
+            self.compile_cell_feature_analysis(file_info, features)
+        self.compile_feature_defs()
+        self.compile_dataset()
 
-    def create_dataset_folder(self):
+    def get_row_data(self, row):
+        """
+        Separates the file info and features from the row
+        """
+        file_info, features = [], []
+        for column, value in row.items():
+            value = self.convert_str_to_num(value)
+            if column in self.FILEINFO_COLUMN_NAMES:
+                file_info.append(value)
+            else:
+                self.process_features(column, value, features)
+        return file_info, features
+
+    def convert_str_to_num(self, value):
+        try:
+            numeric_value = float(value)
+            if numeric_value.is_integer():
+                return int(numeric_value)
+            return numeric_value
+        except ValueError:
+            # if not number, return the original value
+            return value
+
+    def process_features(self, column, value, features):
+        if self.is_valid_feature_value(value):
+            features.append(value)
+            self.update_feature_metadata(column, value)
+        else:
+            print(f"Invalid value: {value} in column {column}")
+            return
+
+    def is_valid_feature_value(self, value):
+        return pd.isna(value) or isinstance(value, (int, float))
+
+    def update_feature_metadata(self, column, value):
+        is_discrete = self.is_discrete(value)
+        self.discrete_features_dict[column] = is_discrete
```

Comment:
Should this be an AND operation? Like, a feature should only be marked as discrete if every value in the column is a discrete value. Currently this will only consider the last row in the CSV, since it gets overwritten each row.

Something like:
```
is_discrete = self.is_discrete(value)
if not column in self.discrete_features_dict:
    self.discrete_features_dict[column] = is_discrete
else:
    self.discrete_features_dict[column] = is_discrete and self.discrete_features_dict[column]
```